### ISSUE TITLE
Real IP when using NPM Streams

### ISSUE THREAD

**Issue by raratdev** [NginxProxyManager/nginx-proxy-manager#3337]
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**
I have an OVH proxmox server with a Debian 11 VM (NAT network).
Basically i simply installed npm in my VM and using Streams i redirected port 9987 UDP
to my home teamspeak ip.
It works, i can connect my OVH VM ip:port and it redirects me to my home teamspeak server.
The problem is that when i check my user ip inside teamspeak, it shows my OVH VM ip.
I want to be able to see my real ip.

**Nginx Proxy Manager Version**
v2.10.4 (latest)

**Expected behavior**
I expect to see my real ip in teamspeak user connection info, not OVH VM ip.

**Operating System**
Debian 11

Any possible solution?

**Comment by raratdev** [comment]
No one can help?

**Comment by fenghuan1994** [comment]
+1，i hava this problem too....

**Comment by senseiriksha** [comment]
+1 same problem

**Comment by raratdev** [comment]
Is this problem still present?

**Comment by senseiriksha** [comment]
Yes and no. Later, I realized that you need to configure headers in NPM to forward the real client IP address to the backend service, which then needs to be configured to correctly handle those headers.

**Comment by epiphany11** [comment]
Yes, it would be very convenient if there were an option for Custom Configuration under the streams section. THX

**Comment by raratdev** [comment]
> Yes and no. Later, I realized that you need to configure headers in NPM to forward the real client IP address to the backend service, which then needs to be configured to correctly handle those headers.

Can you give more details of what to do?
Thanks

**Comment by senseiriksha** [comment]
> > Yes and no. Later, I realized that you need to configure headers in NPM to forward the real client IP address to the backend service, which then needs to be configured to correctly handle those headers.
> 
> Can you give more details of what to do? Thanks

In my case, the issue was not with TeamSpeak itself, but with the Apache server. All traffic appeared to be coming from the same IP address, which caused problems when configuring firewall rules on my website.

To resolve this, I had to add custom headers and explicitly instruct Apache how to handle them so it could correctly determine the real client IP address.

You should check which headers TeamSpeak uses to determine the client IP and try configuring those headers in NPM so the real IP is properly forwarded.

Here is an example from my setup, just to give you an idea of what you might be dealing with:
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
